types/string.go

```diff
@@ -0,0 +1,82 @@
+package types
+
+import (
+	"context"
+
+	tf "github.com/hashicorp/terraform-plugin-framework"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+type StringType struct{}
+
+// TerraformType returns the tftypes.Type that should be used to
+// represent this type. This constrains what user input will be
+// accepted and what kind of data can be set in state. The framework
+// will use this to translate the AttributeType to something Terraform
+// can understand.
+func (s StringType) TerraformType(ctx context.Context) tftypes.Type {
```

Comment:
That's a good question, and I tried to put it where I thought we'd need it for logging, but I think my answer is just going to be "yes, until proven otherwise", with the exception of the `Equal` method, which needs a specific type signature if `go-cmp` is going to be able to use it.

Reply:
Oh, sorry, you were asking the other way, of "can we get rid of it". I think my preference is to keep it here, as it's necessary to get to a logger instance, so anywhere a provider may want to have access to a logger, I want to make sure a context is available.

Reply:
Yes, good point regarding loggers. Context everywhere then